Rpackage/R/bestsplit.R

```diff
@@ -0,0 +1,64 @@
+bestsplit <- structure(function
+### Compute vector of optimal prediction and cost.
+#We have the feature and target value of the data that is to be splitted.
+(target.mat,feature.mat,margin=0.0,loss="hinge"){
+  #We keep track of the following values for each optimal split
+  best_split <- NULL
+  best_split$cost <- Inf
+  best_split$br <- NULL
+  best_split$varid <- NULL
+  best_split$row <- NULL
+
+  #loop for every feature
+  for (index in 2:(length(feature.mat[1,]))){
+    feat <- feature.mat[, index]
+
+    #sorted
+    sorted <- order(feat)
+    z<- feat
+    feat <- z[sorted]
+    z<- target.mat
+    tar <- z[sorted,]
+    rev_tar <- z[rev(sorted),]   #reverse order
+
+    #1st and last index of duplicate elem
+    first_idx <- order(feat)[!duplicated(feat)]
+    last_idx <- c(order(feat)[!duplicated(feat)]-1,length(feat))
```

Comment:
Coding standards: please use spaces after operators and commas.

E.g.:
c(order(feat)[!duplicated(feat)]-1,length(feat)) --> c(order(feat)[!duplicated(feat)] - 1, length(feat))